### PR TITLE
bump native SDKs to 3.1

### PR DESCRIPTION
**Pull request by ianlin-bbpos**
## Summary

bump native SDKs to 3.1.0.

## Motivation

add user cancellation for internet readers.

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one:

- [ ] I have added relevant documentation for my changes.
- [x] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt [hunk 1]
```diff
@@ -537,12 +541,13 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
             }
 
             val customerConsentCollected = getBoolean(params, "customerConsentCollected")
+            val enableCustomerCancellation = getBoolean(params, "enableCustomerCancellation")
```
Comment (chr-stripe) [inline android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:544]: Does `getBoolean()` default to `false` if it's not provided?

**Review by chr-stripe — DISMISSED** [review]
Android changes LGTM 👍

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt (hunk 1 above)
Comment (nazli-stripe) [reply android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:544]: yep https://github.com/stripe/stripe-terminal-react-native/blob/abccbdf4/android/src/main/java/com/stripeterminalreactnative/Mappers.kt#L41-L42

**Review by bric-stripe — DISMISSED** [review]
iOS changes lgtm 👍

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt [hunk 2]
```diff
@@ -4,6 +4,7 @@ import android.annotation.SuppressLint
 import android.app.Application
 import android.content.ComponentCallbacks2
 import android.content.res.Configuration
+import android.util.Log
```
Comment (nazli-stripe) [inline android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:7]: this seems unused ✂️

File: src/types/index.ts [hunk 3]
```diff
@@ -183,10 +183,12 @@ export type CollectPaymentMethodParams = {
...
   tipEligibleAmount?: number;
   updatePaymentIntent?: boolean;
+  enableCustomerCancellation?: boolean;
 };
 
 export type CollectSetupIntentPaymentMethodParams = {
   customerConsentCollected?: boolean;
+  enableCustomerCancellation?: boolean;
   setupIntent: SetupIntent.Type;
 };
```
Comment (mshafrir-stripe) [inline src/types/index.ts:193]: Does order matter?

Also in the other SDKs, we create a new "config" class that contained the `enableCustomerCancellation` param.

https://site-admin.stripe.com/docs/terminal/references/api/js-sdk#collect-setup-intent-payment-method

Comment (mshafrir-stripe) [reply src/types/index.ts:193]: (also 👋 Ian!)

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt (hunk 2 above)
Comment (ianlin-bbpos) [reply android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:7]: got it.

File: src/types/index.ts (hunk 3 above)
Comment (ianlin-bbpos) [reply src/types/index.ts:193]: Hi, Machael, the order doesnt matter, we just need to match them well.  right? @nazli-stripe

**Review by nazli-stripe — APPROVED** [review]

**Pull request merged**